src/golang/lib/workflow/operator/base.go

```diff
@@ -372,6 +372,10 @@ func (bo *baseOperator) PersistResult(ctx context.Context) error {
 			artifactExecState.Status = shared.CanceledExecutionStatus
 		}
 
+		if artifactExecState.Status == shared.SucceededExecutionStatus && !outputArtifact.ShouldPersistContent() {
+			artifactExecState.Status = shared.DeletedExecutionStatus
```

Comment:
This setting of status to Deletion before the artifact is even deleted seems confusing actually, can we not update the status in `DeleteContent()` instead? That way all the deletion logic is handled in one place.

Reply:
In this way the artifact will be marked as succeeded during execution and deleted later. I'm not sure if it's going to confuse users since they will see the intermediate succeed state

Reply:
Yes - I actually find that less confusing since it very clearly follows the lifecycle, where deleted is just a future state of succeeded. Right now, we technically would have operators using artifacts already marked as deleted, which seems weirder?

Reply:
sounds good, let's do what you suggested